File: Android/ProjectoESeminario/app/src/main/java/andre/pt/projectoeseminario/ClipboardControllerFactory.java
package andre.pt.projectoeseminario;

import andre.pt.projectoeseminario.State.ClipboardController;


public class ClipboardControllerFactory {
    private static ClipboardController clipboardController;


    public static ClipboardController getSingleton(String d){
        if(clipboardController == null)
            clipboardController = new ClipboardController(d);

        return clipboardController;
    }

}

File: Windows/ProjectoESeminario/ProjectoESeminario/State/ClipboardControllerFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectoESeminario.State
{
    public class ClipboardControllerFactory
    {
        private static ClipboardController clipboardController;


        public static ClipboardController getSingleton(String data)
        {
            if (clipboardController == null)
                clipboardController = new ClipboardController(data);

            return clipboardController;
        }

    }
}
